tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -3914,9 +3950,183 @@ Status ConvertTopK(OpConverterParams* params) {
+  TRT_ShapedWeights score_threshold = inputs.at(5).weights();
+
+  // Validate tensors and weights (also set some of the needed plugin fields)
+  const auto boxes_dims = boxes_tensor->getDimensions();
+  const auto scores_dims = scores_tensor->getDimensions();
+  if (boxes_dims.nbDims != 3) {
+    return errors::InvalidArgument(
+        "TensorRT BatchedNMS Plugin input boxes must be 3-D excluding batch ",
+        node_def.name());
+  }
+  const int num_classes = scores_dims.d[1];
+  bool box_check = boxes_dims.d[1] == 1 || boxes_dims.d[1] == num_classes;
+  if (!box_check) {
+    return errors::InvalidArgument(
+        "TensorRT BatchedNMS Plugin third dimension of boxes must be either 1 "
+        "or num_classes ",
+        node_def.name());
+  }
+  if (output_size_per_class.shape_.nbDims != 1) {
+    return errors::InvalidArgument(
+        "TensorRT BatchedNMS Plugin max_output_size_per_class must be 0-D ",
+        node_def.name());
+  }
+  int max_size_per_class = *(
+      static_cast<int*>(const_cast<void*>(output_size_per_class.GetValues())));
+  if (max_size_per_class <= 0) {
+    return errors::InvalidArgument(
+        "TensorRT BatchedNMS Plugin max_output_size_per_class should be > 0",
+        node_def.name());
+  }
+  if (total_size.shape_.nbDims != 1) {
+    return errors::InvalidArgument(
+        "TensorRT BatchedNMS Plugin max_total_size must be 0-D ",
+        node_def.name());
+  }
+  int max_total_size =
+      *(static_cast<int*>(const_cast<void*>(total_size.GetValues())));
+  if (max_total_size <= 0) {
+    return errors::InvalidArgument(
+        "TensorRT BatchedNMS Plugin max_total_size should be > 0",
+        node_def.name());
+  }
+  if (iou_threshold.shape_.nbDims != 1) {
+    return errors::InvalidArgument(
+        "TensorRT BatchedNMS Plugin iou_threshold must be 0-D ",
+        node_def.name());
+  }
+  float iou_thresh =
+      *(static_cast<float*>(const_cast<void*>(iou_threshold.GetValues())));
+  if (iou_thresh < 0.0 || iou_thresh > 1.0) {
+    return errors::InvalidArgument(
+        "TensorRT BatchedNMS Plugin iou_threshold must be in [0, 1]",
+        node_def.name());
+  }
+  if (score_threshold.shape_.nbDims != 1) {
+    return errors::InvalidArgument(
+        "TensorRT BatchedNMS Plugin score_threshold must be 0-D ",
+        node_def.name());
+  }
+
+  if (params->validation_only) return Status::OK();
+
+  // Set plugin fields and the field collection
+  TFAttrs attrs(node_def);
+  bool share_location = (boxes_dims.d[1] == 1);
+  const bool pad_per_class = attrs.get<bool>("pad_per_class");
+  int topK;
+  if (pad_per_class) {
+    topK = std::min(max_size_per_class * num_classes, max_total_size);
+  } else {
+    topK = max_total_size;
+  }
+  const int keepTopK = topK;
+  float score_thresh =
+      *(static_cast<float*>(const_cast<void*>(score_threshold.GetValues())));
+  const int background_id = -1;
+  nvinfer1::PluginField fields[7] = {
```

Comment:
@pooyadavoodi Hi Pooya, would you please help to add the `is_normalized` attribute/field you mentioned to fix the accuracy issue? Thanks.